src/runez/pyenv.py

```diff
@@ -952,11 +945,15 @@ def __hash__(self):
 
     def __eq__(self, other):
         other = Version.from_text(other, strict=True)
-        return isinstance(other, Version) and self.components == other.components and self.prerelease == other.prerelease
+        if isinstance(other, Version):
+            return self.epoch == other.epoch and self.components == other.components and self.prerelease == other.prerelease
```

Comment:
Let's add an else clause and return `False` so the return type can be consistently bool.